### PR TITLE
fix(v2): validate hour parameter and warn on catch-up budget exhaustion

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

- **Fix swallowed `strconv.Atoi` errors** in 4 locations in `v2only/datastore.go` where invalid hour strings (e.g., `"abc"`) silently defaulted to 0 (midnight), causing wrong query results
- **Add context-aware API validation**: hourly queries require a single integer 0-23 (returns HTTP 400 otherwise); other query types accept hour ranges via `parseHourFilter()`
- **Add warning log** when migration catch-up exhausts its batch budget without completing a full scan of the legacy database

## Details

From the [v2 datastore code review](docs/plans/2026-02-03-v2-datastore-code-review.md), two high-priority issues:

**Issue 1 — Hour parsing (4 locations):**
- Added `parseHour()` helper with `ErrInvalidHour` sentinel error (defense in depth at datastore layer)
- API layer validates hour parameter based on query type context, preventing both HTTP 500 from invalid input and silent data correctness issues
- Gemini review identified edge cases: empty hour on hourly queries and range values like `"6-9"` that would pass generic validation but fail at `strconv.Atoi`

**Issue 2 — Catch-up budget exhaustion:**
- Added `reachedEnd` flag in `runCatchUp()` to distinguish normal completion from budget exhaustion
- Logs a warning with diagnostic fields (`max_batches`, `batch_size`, `records_caught`) when the loop exits without scanning all records

## Test plan

- [x] 11 unit tests for `parseHour` (valid, invalid, boundary values)
- [x] 4 new API integration tests (invalid hour, missing hour, range rejected, out-of-range)
- [x] All existing tests pass (12/12 `TestGetDetections`, full v2 test suite)
- [x] Linter: 0 issues
- [x] Independent review by both code-reviewer agent and Gemini 3 Pro — both PASS

## Summary by CodeRabbit

* **Bug Fixes**
  * Stricter validation of hour parameters in detection queries: hourly queries now require a single integer 0–23; invalid or missing values return clear 400 errors.
  * Improved catch-up migration logging: warns when a catch-up scan does not complete within the configured batch budget.

* **Tests**
  * Added comprehensive tests covering invalid, missing, out-of-range, and range-format hour inputs for detection queries.

### CODE REVIEW

File: internal/api/v2/detections.go [hunk 1]
```diff
@@ -280,6 +280,22 @@ func (c *Controller) parseDetectionQueryParams(ctx echo.Context) (*detectionQuer
...
+		if params.Hour == "" {
+			return nil, echo.NewHTTPError(http.StatusBadRequest, "hour parameter is required for hourly query type")
+		}
+		if h := parseHourFilter(params.Hour); h == nil || h.Start != h.End {
+			return nil, echo.NewHTTPError(http.StatusBadRequest, "invalid hour parameter: must be a single integer between 0 and 23")
+		}
+	} else if params.Hour != "" {
+		// Other query types: hour is optional, but if present must be valid (ranges OK)
+		if parseHourFilter(params.Hour) == nil {
+			return nil, echo.NewHTTPError(http.StatusBadRequest, "invalid hour parameter: must be an integer between 0 and 23")
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/api/v2/detections.go]: Fixed. Replaced parseHourFilter + Start==End check with direct strconv.Atoi for the hourly path. Now `"08-08"` is correctly rejected as HTTP 400. Added test case for this edge case.

[the remark this answers is not in the document]
Comment (tphakala) [reply internal/api/v2/detections.go]: Fixed. Updated the error message on the non-hourly path to mention that hour ranges are also accepted: "must be a valid hour (0-23) or hour range (e.g. 6-9)".

**Comment by tphakala** [comment]
## Responses to review feedback

**CodeRabbit actionable (range syntax "08-08"):** Fixed in 4e7d35d. Replaced `parseHourFilter` + `Start==End` with direct `strconv.Atoi` for hourly queries. Added test case.

**CodeRabbit nitpick (named constants):** Fixed in 4e7d35d. Added `minHour`/`maxHour` constants.

**CodeRabbit nitpick (WithContext on warning log):** Declining. No other `w.logger.Warn()` call in `worker.go` uses `WithContext(ctx)` (10 existing calls). Adding it only to this one warning would be inconsistent with the rest of the file. If `WithContext` adoption is desired, it should be done file-wide separately.

**Gemini (error message for non-hourly path):** Fixed in 0d2baff. Updated message to mention hour ranges are accepted.
